components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/impl/SystemScopesApiServiceImpl.java

```diff
@@ -54,4 +58,17 @@ public Response systemScopesScopeNameGet(String scope, String username, MessageC
             }
         return Response.ok().entity(scopeSettingsDTO).build();
     }
+
+    public Response systemScopesGet(MessageContext messageContext) throws APIManagementException {
+        try {
+            Map<String, String> scopeRoleMapping = APIUtil.getRESTAPIScopesForTenant(MultitenantUtils
+                    .getTenantDomain(RestApiUtil.getLoggedInUsername()));
+            ScopeListDTO scopeListDTO = SystemScopesMappingUtil.fromScopeListToScopeListDTO(scopeRoleMapping);
+            return Response.ok().entity(scopeListDTO).build();
+        } catch (Exception e) {
+            String errorMessage = "Error when getting the list of scopes-role mapping.";
```

Comment:
Why are we catching a generic exception here?

Reply:
It should be APIManagamentException
corrected with 55a9a47